[Data] Add section on diagnosing reduced throughput with local shuffle buffer (#42615)
ray-project/ray: doc/source/data/transforming-data.rst

    network transfer. This option can be used together with shuffling ordering of
    files.

.. tip::

    You may experience reduced throughput when using ``local_shuffle_buffer_size``;
    one way to diagnose this is to check the total time spent in batch creation by
    examining the ``ds.stats()`` output (``In batch formatting`` under
    ``Batch iteration time breakdown``). If this time is significantly larger than the
    time spent in other steps, one way to improve performance is to decrease

>     ``local_shuffle_buffer_size`` or turn off the local shuffle buffer altogether.

Reviewer: maybe also mention the input file shuffle method as a cheaper alternative?

.. tip::

    This is slower than shuffling ordering of files, and shuffles rows locally without
    network transfer. This local shuffle buffer can be used together with shuffling
    ordering of files; see :ref:`Shuffle the ordering of files <shuffling_file_order>`.

    If you observe reduced throughput when using ``local_shuffle_buffer_size``;
    one way to diagnose this is to check the total time spent in batch creation by
    examining the ``ds.stats()`` output (``In batch formatting``, under
    ``Batch iteration time breakdown``).
    
    If this time is significantly larger than the
    time spent in other steps, one way to improve performance is to decrease

>     ``local_shuffle_buffer_size`` or turn off the local shuffle buffer altogether.
>     You can instead only :ref:`shuffle the ordering of files <shuffling_file_order>`
>     for a computationally cheaper approach.

Reviewer 2: ```suggestion
    ``local_shuffle_buffer_size`` or turn off the local shuffle buffer altogether and only :ref:`shuffle the ordering of files <shuffling_file_order>`.
```

(just shortened it since it's already mentioned at the beginning of the tip.
